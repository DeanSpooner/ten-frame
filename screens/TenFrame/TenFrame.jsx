import { useState } from "react";
import { Text, View, TextInput } from "react-native";
import Table from "../../components/Table/Table";
import tenFrameStyles from "./TenFrame.styles";

const TenFrame = () => {
  const [counters, setCounters] = useState(0);

  const onChangeAmount = (newAmount) => {
    if (
      newAmount > 10 ||
      newAmount.charCodeAt(0) < 48 ||
      newAmount.charCodeAt(0) > 57 ||
      newAmount.charCodeAt(1) < 48 ||
      newAmount.charCodeAt(1) > 57
    ) {
      return;
    }
    setCounters(newAmount);
  };

  return (
    <>
      <Text style={tenFrameStyles.title}>Ten Frame</Text>
      <Text style={tenFrameStyles.text}>
        How many counters would you like to place?
      </Text>
      <View style={tenFrameStyles.inputContainer}>
        <TextInput
          style={tenFrameStyles.textInput}
          onChangeText={(newAmount) => onChangeAmount(newAmount)}
          returnKeyType="done"
          keyboardType="number-pad"
          value={counters}
          maxLength={2}
          contextMenuHidden={true}
        ></TextInput>
      </View>
      <Table counter={counters} />
    </>
  );
};

export default TenFrame;
