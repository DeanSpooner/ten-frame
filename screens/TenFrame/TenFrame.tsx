import React, { useState } from "react";
import { Text, View, TextInput } from "react-native";
import Table from "../../components/Table/Table";
import tenFrameStyles from "./TenFrame.styles";

const TenFrame: React.FC<{}> = () => {
  const [counters, setCounters] = useState<number>(0);

  //@ts-ignore
  const onChangeAmount = (newAmount) => {
    if (
      newAmount > 10 ||
      newAmount.charCodeAt(0) < 48 ||
      newAmount.charCodeAt(0) > 57 ||
      newAmount.charCodeAt(1) < 48 ||
      newAmount.charCodeAt(1) > 57
    ) {
      return;
    }
    setCounters(newAmount);
  };

  return (
    <>
      <Text style={tenFrameStyles.text}>How many counters will you place?</Text>
      <View style={tenFrameStyles.inputContainer}>
        <TextInput
          style={tenFrameStyles.textInput}
          onChangeText={(newAmount) => onChangeAmount(newAmount)}
          returnKeyType="done"
          keyboardType="number-pad"
          //@ts-ignore
          value={counters}
          maxLength={2}
          contextMenuHidden={true}
        ></TextInput>
      </View>
      <Table counter={counters} />
    </>
  );
};

export default TenFrame;
