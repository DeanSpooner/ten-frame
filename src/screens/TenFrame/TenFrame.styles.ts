import { StyleSheet } from "react-native";

const tenFrameStyles = StyleSheet.create({
  textInput: {
    height: 60,
    margin: 12,
    width: "100%",
    padding: 10,
    fontSize: 40,
    textAlign: "center",
  },
  inputContainer: {
    backgroundColor: "#fff",
    width: "25%",
    height: 60,
    marginVertical: 10,
    borderRadius: 10,
    alignItems: "center",
    justifyContent: "center",
  },
});

export default tenFrameStyles;
