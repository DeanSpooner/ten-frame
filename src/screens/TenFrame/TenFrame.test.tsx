import React from "react";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import TenFrame from "./TenFrame";

describe("<TenFrame />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<TenFrame />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("contains 3 direct child components", () => {
    const tree = renderer
      .create(<TenFrame />)
      .toJSON() as ReactTestRendererJSON[];
    expect(tree?.length).toBe(3);
  });
});
