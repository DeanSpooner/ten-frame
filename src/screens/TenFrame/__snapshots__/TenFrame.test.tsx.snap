// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TenFrame /> renders correctly 1`] = `
[
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 24,
        "marginHorizontal": 5,
        "textAlign": "center",
      }
    }
  >
    How many counters will you place?
  </Text>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 10,
        "height": 60,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "25%",
      }
    }
  >
    <TextInput
      contextMenuHidden={true}
      keyboardType="number-pad"
      maxLength={2}
      onChangeText={[Function]}
      returnKeyType="done"
      style={
        {
          "fontSize": 40,
          "height": 60,
          "margin": 12,
          "padding": 10,
          "textAlign": "center",
          "width": "100%",
        }
      }
      value={0}
    />
  </View>,
  <View
    style={
      {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "maxWidth": 360,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 72,
          "justifyContent": "center",
          "width": 72,
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 72,
          "justifyContent": "center",
          "width": 72,
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 72,
          "justifyContent": "center",
          "width": 72,
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 72,
          "justifyContent": "center",
          "width": 72,
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 72,
          "justifyContent": "center",
          "width": 72,
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 72,
          "justifyContent": "center",
          "width": 72,
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 72,
          "justifyContent": "center",
          "width": 72,
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 72,
          "justifyContent": "center",
          "width": 72,
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 72,
          "justifyContent": "center",
          "width": 72,
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 72,
          "justifyContent": "center",
          "width": 72,
        }
      }
    >
      <View
        style=""
      />
    </View>
  </View>,
]
`;
